packages/open-truss/src/components/OTRestDataProvider.tsx

```diff
@@ -0,0 +1,104 @@
+import { mapValues, template } from 'lodash'
+
+import {
+  withChildren,
+  BaseOpenTrussComponentV1PropsShape,
+  type BaseOpenTrussComponentV1,
+} from '../configuration/engine-v1'
+import {
+  NumberSignal,
+  StringSignal,
+  UnknownSignal,
+  signalValueShape,
+  useSignalEffect,
+} from '../signals'
+import { z } from 'zod'
+
+interface SynchronousRestResult {
+  status: number
+  headers: Record<string, string[]>
+  body: any
+}
+
+const StringOrSignal = z.union([z.string(), StringSignal])
+
+export const Props = BaseOpenTrussComponentV1PropsShape.extend({
+  ...withChildren,
+  source: z.string(),
+  path: StringOrSignal,
+  method: StringOrSignal,
+  headers: z.record(StringOrSignal).optional(),
+  path_values: z.record(StringSignal).optional(),
+  force_query: NumberSignal,
+  output: z.array(UnknownSignal).optional(),
+})
+
+const OTRestDataProvider: BaseOpenTrussComponentV1<z.infer<typeof Props>> = (
+  props,
+) => {
+  const {
+    source,
+    path: templatePath,
+    path_values, // eslint-disable-line @typescript-eslint/naming-convention
+    method,
+    headers,
+    force_query, // eslint-disable-line @typescript-eslint/naming-convention
+    children,
+    output,
+    _DEBUG_,
+  } = props
+
+  useSignalEffect(() => {
+    if (_DEBUG_)
+      console.log({
+        m: 'REST values',
+        source,
+        path: templatePath,
+        method,
+        headers,
+      })
+
+    const stringifiedPathValues = mapValues(path_values, String)
+    const resolvedPath = template(String(templatePath))(stringifiedPathValues)
+
+    let queryResults: SynchronousRestResult
+    const fetchData = async function (): Promise<undefined> {
+      const result = await fetch('/api/synchronous-rest-query', {
```

Comment:
Could this endpoint address be configurable?

Reply:
Good question! It's not documented but my understanding is it's not supposed to be. My understanding from talking with @hktouw is that `open-truss` servers must define an endpoint with this path to be compatible with `open-truss` data providers.

Reply:
I believe the thinking is the backend should define the sources + credentials so we only need to have one url. That being said, this url doesn't have to be `/api/synchronous-rest-query` so perhaps we can make this default to that and have it overridable via an argument.

As an aside, we've talked about supporting overrides for default prop values someday, so in this case you can set a set a new default instead of `/api/synchronous-rest-query`. That would make it convenient if your app has a different endpoint name than the one hard coded here. But we'll get to that someday. :)

Reply:
Cool yea I have no immediate plans to use this but I can't have the url be this so if/when we adopt this I'll update this to be configurable